spec/rubocop/cop/style/prefer_interpolation_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::PreferInterpolation do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense and corrects for string concatenation' do
+    expect_offense(<<~RUBY)
+      email_with_name = user.name + ' <' + user.email + '>'
+                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer string interpolation instead of string concatenation.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      email_with_name = "\#{user.name} <\#{user.email}>"
+    RUBY
+  end
+
+  it 'registers an offense and corrects for string concatenation as part of other expression' do
+    expect_offense(<<~RUBY)
+      users = (user.name + ' ' + user.email) * 5
+               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer string interpolation instead of string concatenation.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      users = ("\#{user.name} \#{user.email}") * 5
```

Comment:
Would be nice to drop these outter `()` too, but maybe a different cop should do this?